.github/workflows/release.yml

```diff
@@ -30,28 +34,28 @@ jobs:
           git config user.email "actions@github.com"
           git config user.name "GitHub Actions"
  
- #     - name: Setup Git user
- #       uses: fregante/setup-git-user@v2 
- #       
-#      - name: Setup SSH Keys and known_hosts
-#        env:
-#          SSH_AUTH_SOCK: /tmp/ssh_agent.sock
-#        run: |
-#            mkdir -p ~/.ssh
-#            ssh-keyscan github.com >> ~/.ssh/known_hosts
-#            ssh-agent -a $SSH_AUTH_SOCK > /dev/null
-#            ssh-add - <<< "${{ secrets.SSH_PRIVATE_KEY }}"
+      - name: Set Release Version
+        id: vars
+        shell: bash
+        run: |
+          echo "VERSION=${{ github.event.inputs.version }}" >> $GITHUB_OUTPUT
+          mvn versions:set -DnewVersion=${{ github.event.inputs.version }}-SNAPSHOT
+          git commit -m "Releasing version ${{ github.event.inputs.version }}" pom.xml
+          git push origin main
         
       - name: Publish to GitHub Packages
         run: mvn -ntp -B release:prepare release:perform
         env:
           GITHUB_TOKEN: ${{ github.token }}
 
-      # Create a release
+      # Create a GitHub Release
+      # TODO (oleg-nenashev): Consider switching to Release Drafter if there is not significant value except fancy changelog
```

Comment:
FWIW release-drafter works with issues/PRs while JReleaser works with commit messages (though it can also invoke GiHub's "native" release notes generator, which also only works with issues/PRs).

Reply:
Exactly. Both approaches have their pros and cons, though I got more used to the PR-based approach. I wanted to try out JReleaser on something in any case, at least until we have unified labeling policy and changelog format in the organization, similar to https://github.com/jenkinsci/.github/blob/master/.github/release-drafter.yml